src/backend/utils/misc/guc_gp.c

```diff
@@ -3215,6 +3216,16 @@ struct config_int ConfigureNamesInt_gp[] =
 		NULL, NULL, NULL
 	},
 
+	{
+		{"gp_random_insert_segments", PGC_USERSET, CUSTOM_OPTIONS,
+			gettext_noop("Use limited number of segments for random distributed table insertion."),
+			NULL
+		},
+		&gp_random_insert_segments,
+		0, 0, INT_MAX,
```

Comment:
What happens if the `gp_random_insert_segments` is 0 or greater than the number of segments in the cluster?

Reply:
0 means that the feature is disabled, greater than actual segments number will have any effect.